src/overlays/actors/ovl_En_Hg/z_en_hg.c

```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
+    /* Powder Keg     */ DMG_ENTRY(1, 0xF),
+};
+
+static CollisionCheckInfoInit2 sColChkInfoInit2 = {
+    0, 0, 0, 0, 0x80,
+};
+
+static ActorAnimationEntry animations[] = {
+    { &D_0600260C, 1.0f, 0.0f, 0.0f, 0x00, -4.0f }, { &D_06009D44, 1.0f, 0.0f, 0.0f, 0x00, -4.0f },
+    { &D_0600A164, 1.0f, 0.0f, 0.0f, 0x00, -4.0f }, { &D_06000370, 1.0f, 0.0f, 0.0f, 0x02, 0.0f },
+    { &D_06001138, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },  { &D_060015D4, 1.0f, 0.0f, 0.0f, 0x02, 0.0f },
+    { &D_06001960, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },  { &D_0600AE1C, 1.0f, 0.0f, 0.0f, 0x00, 0.0f },
+};
+
+u32 D_80BD00C8 = 0;
+
+void EnHg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHg* this = THIS;
+    s16 phi_s1;
+    s32 i;
+
+    phi_s1 = this->actor.cutscene;
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 36.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06008580, &D_0600260C, this->limbDrawTbl, this->transitionDrawTbl,
+                     0x13);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit2);
+    if ((gSaveContext.weekEventReg[0x4B] & 0x20) || (gSaveContext.weekEventReg[0x34] & 0x20)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actor.targetMode = 1;
+    this->actor.colChkInfo.health = 0;
+    this->actor.gravity = -1.0f;
+    for (i = 0; i < 4; i++) {
+        if (phi_s1 == -1) {
+            break;
+        }
+        this->cutscenes[i] = phi_s1;
+        phi_s1 = ActorCutscene_GetAdditionalCutscene(phi_s1);
+    }
+    func_80BCF354(this);
+}
+
+void EnHg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHg* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BCF354(EnHg* this) {
+    func_800BDC5C(&this->skelAnime, animations, 0);
+    this->actionFunc = func_80BCF398;
+}
+
+void func_80BCF398(EnHg* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health == 1) {
+        if ((this->actor.xzDistToPlayer < 200.0f && this->actor.yDistToPlayer < 40.0f) &&
+            (func_800EE29C(globalCtx, 0x1E3U) == 0)) {
+            func_80BCF468(this);
+        }
+        if ((gSaveContext.sceneSetupIndex == 0 && globalCtx->csCtx.unk_12 == 0) &&
+            (globalCtx->csCtx.frames == 0x14 || globalCtx->csCtx.frames == 0x3C)) {
+            Audio_PlayActorSound2(&this->actor, 0x3AB8);
+        }
+    }
+}
+
+void func_80BCF468(EnHg* this) {
+    func_800BDC5C(&this->skelAnime, animations, 1);
+    this->actionFunc = func_80BCF4AC;
+}
+
+void func_80BCF4AC(EnHg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/EnHg_Destroy.asm")
+    this->actor.speedXZ = 1.6f;
+    if (!(player->stateFlags2 & 0x08000000)) {
+        if (func_80152498(&globalCtx->msgCtx) == 0) {
```

Comment:
```suggestion
        if (!func_80152498(&globalCtx->msgCtx)) {
```

Reply:
This if could probably be merged with the one above with an `&&`